examples/tree-view/actions/index.js

```diff
@@ -0,0 +1,26 @@
+export const INCREMENT = 'INCREMENT'
+export const CREATE_NODE = 'CREATE_NODE'
+export const ADD_CHILD = 'ADD_CHILD'
+
+export function increment(nodeId) {
+  return {
+    type: INCREMENT,
+    nodeId
+  }
+}
+
+let nextId = 0
+export function createNode() {
+  return {
+    type: CREATE_NODE,
+    nodeId: `new_${nextId++}`
+  }
+}
+
+export function addChild(nodeId, childId) {
```

Comment:
Oh, I got it now :sweat: 

When you call `const childId = createNode().nodeId` you're actually dispatching the action because of `mapDispatchToProps`.

When reading the code, I assumed it was being called only to get the new id.

You were right, it's hard to discuss hypothetical code :-)

Sorry for asking without testing first, and thank you so much for all the good work you're doing :smile: